### PR TITLE
Candidate bidi stream types extension

### PR DESCRIPTION
**Pull request by LPardue**
This PR speaks to comments I made on #71 and #81. It's intended to sketch out how HTTP/3 could be extended to support bidirectional stream type signalling with a fallback to standard HTTP/3 handling (Section 6.1 of RFC 9114 behaviour).

This could be taken further to define the same stream type value for the uni and bid cases, and expand on  error handling etc

### CODE REVIEW

**Review by marten-seemann — APPROVED** [review]

File: draft-ietf-webtrans-http3.md [hunk 1]
```diff
@@ -195,55 +196,54 @@ with an H3_ID_ERROR error code.
...
-  0                   1                   2                   3
-  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                           0x41 (i)                          ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                        Session ID (i)                       ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                         Stream Body                         ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+WebTransport Unidirectional Stream Header {
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-http3.md:235]:
```suggestion
WebTransport Bidirectional Stream Header {
```

File: draft-ietf-webtrans-http3.md [hunk 2]
```diff
@@ -195,55 +196,54 @@ with an H3_ID_ERROR error code.
...
-`WEBTRANSPORT_STREAM` (type=0x41).  The format of the frame SHALL be the frame
-type, followed by the session ID, encoded as a variable-length integer,
-followed by the user-specified stream data ({{fig-bidi-client}}).  The frame
-SHALL last until the end of the stream.
+HTTP/3 reserves client-initiated bidirectional streams for request and
+responses, and does not use server-initiated bidirectional streams; see
+{{Section 6.1 of HTTP3}}.
+
+WebTransport extends HTTP/3 in order to permit endpoints to explicitly declare
+the purpose of bidirectional streams. This is done by sending a stream type,
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-http3.md:220]: It took me a second to grok this, so I'd suggest clarifying the fact that WebTransport is now using HTTP/3's unidirectional stream types for bidirectional streams. Luckily the IANA registry is just called "HTTP/3 Stream Types" so we're good there but making this clearer would help.

File: draft-ietf-webtrans-http3.md [hunk 3]
```diff
@@ -195,55 +196,54 @@ with an H3_ID_ERROR error code.
...
-{: #fig-bidi-client title="WEBTRANSPORT_STREAM frame format"}
+{: #fig-bidi-client title="Bidirectional WebTransport Stream Header"}
 
-HTTP/3 does not by itself define any semantics for server-initiated
-bidirectional streams.  If WebTransport setting is negotiated by both
-endpoints, the syntax of the server-initiated bidirectional streams SHALL be
-the same as the syntax of client-initated bidirectional streams, that is, a
-sequence of HTTP/3 frames.  The only frame defined by this document for use
-within server-initiated bidirectional streams is WEBTRANSPORT_STREAM.
+Note that in order to avoid confusion between bidirectional stream types and
```
Comment (DavidSchinazi) [inline draft-ietf-webtrans-http3.md:242]: Now I'm confused, which of these is true?
1. we are using unidirectional stream types for both uni and bidi streams
2. we are creating a separate concept of bidi stream types

**Review by DavidSchinazi — COMMENTED** [review]
(no hats) The overall shape here seems odd to me: if client-initiated bidi streams now start with a stream type, then we should define an "HTTP Request" stream type, and require that clients send that before their HTTP frames. We also need to make sure that no frames types can ever conflict with stream types otherwise there is confusion?

**Comment by DavidSchinazi** [comment]
Chair: declaring this PR as overcome by events and closing

**Pull request closed without merging**